WcaOnRails/app/webpacker/components/RolesTab/index.jsx

```diff
@@ -6,77 +6,133 @@ import {
   List,
 } from 'semantic-ui-react';
 import useLoadedData from '../../lib/hooks/useLoadedData';
+import useSaveAction from '../../lib/hooks/useSaveAction';
 import {
   roleListUrl,
+  currentUserUrl,
+  teamUrl,
 } from '../../lib/requests/routes.js.erb';
 import Errored from '../Requests/Errored';
 import Loading from '../Requests/Loading';
 import RoleForm from './RoleForm';
+import I18n from '../../lib/i18n';
 
-function RoleFormModal({
-  trigger, userId, isActiveRole, sync,
-}) {
-  const [open, setOpen] = React.useState(false);
-
-  return (
-    <Modal
-      size="fullscreen"
-      onClose={() => {
-        setOpen(false);
-        sync();
-      }}
-      onOpen={() => setOpen(true)}
-      open={open}
-      trigger={trigger}
-    >
-      <Modal.Content>
-        <RoleForm
-          userId={userId}
-          isActiveRole={isActiveRole}
-        />
-      </Modal.Content>
-    </Modal>
-  );
-}
+// let i18n-tasks know the key is used
+// i18n-tasks-use t('enums.user.role_status.trainee_delegate')
+// i18n-tasks-use t('enums.user.role_status.candidate_delegate')
+// i18n-tasks-use t('enums.user.role_status.delegate')
+// i18n-tasks-use t('enums.user.role_status.senior_delegate')
+// i18n-tasks-use t('enums.user.role_status.member')
+// i18n-tasks-use t('enums.user.role_status.senior_member')
+// i18n-tasks-use t('enums.user.role_status.leader')
 
 export default function RolesTab({ userId }) {
+  const currentUser = useLoadedData(currentUserUrl);
   const {
     data, sync, loading, error,
```

Comment:
Since there are multiple fetches happening, it might be best to rename these to `userRoleData`, etc., or just name the whole const `userRoles` and access them via `roles.data`, etc.? Or maybe `profileRolesData`/`profileRoles` to try and avoid confusion between the user being edited and the current user? Currently this component a bit hard to understand based on the names of things.

Reply:
I've renamed couple of them by appending `fetch` at end. Hope it's more clear now. 🙂